compiler/ballerina-lang/src/test/java/org/wso2/ballerinalang/compiler/util/NumericLiteralSupportTest.java

```diff
@@ -0,0 +1,54 @@
+package org.wso2.ballerinalang.compiler.util;
+
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class NumericLiteralSupportTest {
+
+  @Test
+  public void testStripDiscriminator() {
+    Assert.assertEquals("3", NumericLiteralSupport.stripDiscriminator("3"));
+    Assert.assertEquals("abc", NumericLiteralSupport.stripDiscriminator("abc"));
+    Assert.assertEquals("abc",
+        NumericLiteralSupport.stripDiscriminator("abcf"));
+    Assert.assertEquals("abc",
+        NumericLiteralSupport.stripDiscriminator("abcF"));
+    Assert.assertEquals("abc",
+        NumericLiteralSupport.stripDiscriminator("abcd"));
+    Assert.assertEquals("abc",
+        NumericLiteralSupport.stripDiscriminator("abcD"));
+  }
+
+  @Test
+  public void testIsHexLiteral() {
+    Assert.assertTrue(NumericLiteralSupport.isHexLiteral("p\u0000p"));
+    Assert.assertTrue(NumericLiteralSupport.isHexLiteral("P\u0000P"));
+    Assert.assertFalse(NumericLiteralSupport.isHexLiteral("1"));
+  }
+
+  @Test
+  public void testParseBigDecimal() throws Exception {
+    Assert.assertEquals("1.0",
+        NumericLiteralSupport.parseBigDecimal(1d).toString());
+    Assert.assertEquals("1.0",
+        NumericLiteralSupport.parseBigDecimal(1D).toString());
+    Assert.assertEquals("1",
+        NumericLiteralSupport.parseBigDecimal(1).toString());
```

Comment:
Pass a value with a length longer than 2 without a `d` init
Maybe a alphanumeric ie `b1r` and a numeric value

Reply:
I'm testing for that functionality within the next test for `isDecimalDiscriminated`